Test autobind cleanup

nicolassenechal opened the pull request:
## What does this pull request do?
It removes the dependency on React __reactAutoBindMap in tests, testing instead the output of the functions. It also incidentally makes the tests easier to read.

AlanFoster commented on src/calendar/tests/CalendarSelection.spec.js:
```diff
@@ -5,23 +5,13 @@ import CalendarSelection from '../CalendarSelection';
-      },
-    });
-    expect(this.renderedComponent.props.className).toEqual('my-class');
+    expect(this.renderedComponent.props.className).toContain('CalendarSelection');
```
:question: Would we not expect this to be prefixed prefixed with `DateRangePicker` ?

nicolassenechal replied:
Given that I set the bemBlock attribute to DateRangePicker, it would make sense to test against an actual class name rather than a partial name indeed. I will change it.

AlanFoster replied:
Brilliant, thanks!

AlanFoster commented on src/tests/DateRangePicker.spec.js:
```diff
@@ -340,43 +335,44 @@ describe('The DateRangePicker component', function () {
-        var spy = spyOn(DateRangePicker.prototype.__reactAutoBindMap, 'changeYear');
-        this.useDocumentRenderer();
+        this.useDocumentRenderer({
+          initialYear: 2000,
```
Should we verify `expect(this.renderedComponent.state.year).toBe(2000);` here?

nicolassenechal replied:
I do not think so. This would test that a prop is correctly applied initially to the component state. Looks this might have been overlooked in the initial implementation :)

AlanFoster replied:
:+1:

AlanFoster commented on src/tests/PaginationArrow.spec.js:
```diff
@@ -34,31 +32,10 @@ describe('The Pagination Arrow component', function () {
     expect(this.renderedComponent).toEqual(
-      <div className='my-class' onClick={clickTrigger}>
-        <div className='PaginationArrowIcon' />
+      <div bemBlock='DateRangePicker' className='DateRangePicker__PaginationArrow DateRangePicker__PaginationArrow--next' onClick={clickTrigger} >
```
:grey_question: Should `bemBlock` be here?

nicolassenechal replied:
There is a small issue with the bem functionality. If we don't specify the attribute, we end up with class names such as 'null__PaginationArrowIcon'. I thought this would be a bit cleaner

AlanFoster commented on src/tests/PaginationArrow.spec.js:
```diff
@@ -34,31 +32,10 @@ describe('The Pagination Arrow component', function () {
       </div>
     );
   });
 
```
Let's nuke this whitespace too :+1:

nicolassenechal replied:
But I like whitespaces at the end of self-closing XML tags :) Is there a way to set-up a linting rule for this?

AlanFoster replied:
The whitespace at the end the `it` block encapsulating `describe` block can be nuked IMO; However you're right, there is no linting rule for this. Happy for you to punt on this.

AlanFoster commented:
Awesome PR @nicolassenechal, I only have some minor questions - thanks for the hard work!

nicolassenechal commented:
Sorry for the delay coming back, busy holidaying and relocating :)

Changed the test per the comment (and also fixed a failing test)
For the white space, did you mean the carriage return? I removed it but as for setting a linting rule, I am not sure what it would be and how many extra changes would be required. It feels like it is out the scope of this PR to be fair.

AlanFoster commented:
thanks for the great work @nicolassenechal !
